pkg/phlaredb/profile_test.go

```diff
@@ -279,3 +279,67 @@ func TestProfileIndex_Add_OutOfOrder(t *testing.T) {
 	require.Equal(t, []int64{20, 50, 80, 100, 110}, tsOrder)
 
 }
+
+func Test_rowRangesWithSeriesIndex_getSeriesIndex(t *testing.T) {
+	testCases := []struct {
+		name        string
+		ranges      rowRangesWithSeriesIndex
+		rowNum      int64
+		searchHint  int
+		expectIdx   int
+		expectPanic bool
+	}{
+		{
+			name: "main loop hit",
+			ranges: rowRangesWithSeriesIndex{
+				{rowRange: &rowRange{rowNum: 0, length: 5}, seriesIndex: 1},
+				{rowRange: &rowRange{rowNum: 5, length: 5}, seriesIndex: 2},
+			},
+			rowNum:     6,
+			searchHint: 1,
+			expectIdx:  1,
```

Comment:
there is something wrong with the testate. rownum should point to row range with index 1 with seriesIndex 2

Reply:
we should expect seriesIndex2 here not 1